qiita_db/study.py

```diff
@@ -166,6 +166,18 @@ def status(self):
             return qdb.util.infer_status(
                 qdb.sql_connection.TRN.execute_fetchindex())
 
+    @staticmethod
+    def all_data_types():
+        """Returns list of all the data types available in the system
+        Returns
+        -------
+        list of str
+        """
+        with qdb.sql_connection.TRN:
+            sql = "SELECT DISTINCT data_type FROM qiita.data_type"
+            qdb.sql_connection.TRN.add(sql)
+            return qdb.sql_connection.TRN.execute_fetchflatten()
```

Comment:
Seems like this could return None, should this be checking and instead returning an empty list? Otherwise it would probably be worth updating the docstring. My preference would be to return an empty list, not sure what others think about that.